### PR TITLE
GH-31809: [Docs] Add instructions on how to collect the produced telemetry data

### LINKED ISSUE #31809: [Tools][Docs] Add instructions on how to collect the produced telemetry data
With the ongoing efforts to produce telemetry data in the engine, it might be helpful to collect some notes from the landed PRs about how to collect the telemetry data using the otel-collector and Jaeger running in some containers with docker-compose.

The goal is to make it trivial with these notes to get started with the telemetry data produced by the engine.

**Reporter**: [Matthijs Brobbel](https://issues.apache.org/jira/browse/ARROW-16435) / @mbrobbel
**Assignee**: [Bryce Mecum](https://issues.apache.org/jira/browse/ARROW-16435) / @amoeba

<sub>**Note**: *This issue was originally created as [ARROW-16435](https://issues.apache.org/jira/browse/ARROW-16435). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

### PR DESCRIPTION
**Pull request by amoeba**
### Rationale for this change

The codebase has been partially implemented with the OpenTelemetry SDK and it would be beneficial for users to have a concise guide on how to get started collecting and visualizing tracing data.

### What changes are included in this PR?

This is an initial draft for review of an addition to the [C++ User Guide](https://arrow.apache.org/docs/dev/cpp/user_guide.html) as mentioned in https://github.com/apache/arrow/issues/31809#issuecomment-1378097635.

### Are these changes tested?

These changes have been tested locally by building the docs and verifying.

### Are there any user-facing changes?

Yes, but just documentation changes.
* Closes: #31809

### CODE REVIEW

**Comment by amoeba** [comment]
I've published a built version of the new page in the docs this PR creates to https://63d1d0cd7dae342ddeb4db1a--arrow-docs.netlify.app/cpp/opentelemetry.html for easier viewing.

For whoever reviews, I think a very high-level review would be a good first step. Is this the right shape? Is this in the right location in the docs? I tried to match the style of other user guides in terms of structure and style and avoided writing this up in the more pedagogical style of https://github.com/amoeba/arrow-opentelemetry-example.

**Comment by joosthooz** [comment]
Hi, I think this is a really great step because it allows users to at least get started with tracing! At some point we may want to explain some more about how to read the traces. But the telemetry code in Arrow is currently still in flux so there might be some changes coming up wrt the structure and naming of spans. Thanks for this. Maybe @mbrobbel would like to have a look?

File: docs/source/cpp/opentelemetry.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+To enable tracing, specify ``ARROW_WITH_OPENTELEMETRY=ON`` when generating your
+build. You may also want to specify ``CMAKE_BUILD_TYPE=RelWithDebInfo`` in order
+to get representative timings while retaining debug information.
+
+Exporting Tracing Information
+-----------------------------
+
+By default, no tracing information is exported until a tracing backend has been
+specified. The choice of tracing backend is controlled with the
+:envvar:`_ARROW_TRACING_BACKEND` environment variable. Possible values are:
```
Comment (mbrobbel) [inline docs/source/cpp/opentelemetry.rst:44]:
```suggestion
:envvar:`ARROW_TRACING_BACKEND` environment variable. Possible values are:
```

File: docs/source/cpp/opentelemetry.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+   - ``ostream``: emit textual log messages to stdout
+   - ``otlp_http``: emit JSON traces to a HTTP server (by default, the endpoint
```
Comment (mbrobbel) [inline docs/source/cpp/opentelemetry.rst:47]:
```suggestion
   - ``otlp_http``: emit OTLP JSON encoded traces to a HTTP server (by default, the endpoint
```
The payload for `oltp/http` can be either JSON or binary. We are using the default of the exporter provided by (opentelemetry-cpp) which is JSON.

(you may need to rewrap this suggestion)

**Review by mbrobbel — COMMENTED** [review]
This is great! I added two minor suggestions.

I agree with @joosthooz that we should defer documenting the trace structure until that  stabilizes.

**Comment by amoeba** [comment]
Thanks for taking a look at this @joosthooz and @mbrobbel.

@mbrobbel I accepted both of your suggestions, re-wrapped your one change as you mentioned. I also replicated the changes to the the list of options for ARROW_TRACING_BACKEND in env_vars.rst so both pages are identical.

An updated preview build of the docs is available at https://63d2f00e3c06c61b7852ce4a--tranquil-faloodeh-d58aad.netlify.app/cpp/opentelemetry.html. This is ready for another look and/or a merge.

**Review by mbrobbel — APPROVED** [review]

**Comment by amoeba** [comment]
Hi @mbrobbel, do you want to look at this again or is this ready for a merge?

**Review by lidavidm — APPROVED** [review]

**Comment by amoeba** [comment]
Thank you @lidavidm!

**Pull request merged**